### PR TITLE
feat(bat): support head-pruned embedding-only v2.4 embedding model

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds support for a head-pruned, 1-output "embedding-only" BattyBirdNET v2.4 embedding backbone that carries only the `[1024]` embedding output, dropping the unused `[6522]` BirdNET logits head (about 26 MB smaller). The existing 2-output backbone (logits at output 0, embedding at output 1) keeps working unchanged, so existing installs are unaffected.

birdnet-go previously read the bat embedding by a fixed output index and detected the model by output count, both of which break for a 1-output embedding-only model (a 1-output v2.4 model was assumed to be the plain `[6522]` classifier). This makes those paths shape-aware.

## Changes

- `internal/inference/onnx/detection.go`: a single `[.,1024]` output is configured as embedding-only (`LogitsIndex = -1`, `EmbeddingIndex = 0`, `EmbeddingSize = 1024`), disambiguated from the plain 1-output `[.,6522]` classifier by output size. New `DetectEmbeddingOutput` / `selectEmbeddingOutput` report the embedding output port from model metadata.
- `internal/inference/onnx/classifier.go`: the ORT extraction (`PredictRawWithEmbeddings`), output-tensor sizing (`outputShape`), label validation, and the `Predict`/`PredictBatch` path all handle a model with no logits without indexing an out-of-range logits output.
- `internal/inference/onnx.go`: thin `DetectEmbeddingOutput` wrapper on the inference facade.
- `internal/classifier/bat_onnx.go`: the OpenVINO path derives the embedding output port at load time (index 1 for the 2-output model, 0 for the pruned one) instead of a hardcoded constant, gated behind the existing OpenVINO eligibility check, and falls back to ONNX Runtime on any detection failure. The ORT path derives the port internally via the model config, so it needs no index passed.

Both backends produce the same embedding for the current 2-output model. The model catalog is intentionally not switched in this PR; that follows once the pruned model is published.

## Test Plan

- [x] `go build ./internal/...`, `go vet ./internal/...`, and `go vet -tags openvino ./internal/classifier/` all pass.
- [x] `golangci-lint run` clean on the affected packages.
- [x] New unit tests (`detection_test.go`) cover `buildModelConfig`, `validateLabelCount`, `outputShape`, `processOutput`, and `selectEmbeddingOutput` across the 2-output, head-pruned, and plain-classifier layouts; pass with `-race`.
- [x] New model-gated functional test (`embedding_only_functional_test.go`) exercises the real ORT inference paths on a pruned model (skipped in CI without `ONNX_TEST_EMB_ONLY_MODEL`).
- [ ] Native arm64 validation on rpi5 for both ORT and OpenVINO once the pruned model is produced (embedding parity and end-to-end bat top-k identical to the current 2-output model; embedding stays fp32).

### CODE REVIEW

File: internal/inference/onnx/classifier.go [hunk 1]
```diff
@@ -183,7 +183,12 @@ func resolveModelType(cfg *classifierConfig, inputShapes [][]int64, numOutputs i
 
 // validateLabelCount checks that the label count matches the model's logits output dimension.
 func validateLabelCount(modelCfg *ModelConfig, outputInfos []ort.InputOutputInfo, labelCount int) error {
-	if modelCfg.LogitsIndex < 0 || modelCfg.LogitsIndex >= len(outputInfos) {
+	// An embedding-only model (LogitsIndex < 0) produces no logits, so there is no
+	// logits dimension to validate the label count against.
+	if modelCfg.LogitsIndex < 0 {
+		return nil
+	}
+	if modelCfg.LogitsIndex >= len(outputInfos) {
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx/classifier.go:194]: Good call, fixed in ba18e461. validateLabelCount now fails fast with a clear error when an embedding-only model (LogitsIndex < 0) is loaded with label validation enabled, instead of letting it construct and only error later at prediction time. This also makes it consistent with the existing label/logits count-mismatch rejection. The bat pipeline loads embedding-only models with WithSkipLabelValidation, so that path bypasses this check and is unaffected.

File: internal/inference/onnx.go [hunk 2]
```diff
@@ -113,6 +113,16 @@ func (c *onnxClassifier) Close() {
...
 
+// DetectEmbeddingOutput returns the output-port index and size of a model's BirdNET
+// v2.4 embedding output ([.,1024]): index 1 for the 2-output backbone (logits@0 +
+// embedding@1), index 0 for the head-pruned, embedding-only model. The bat OpenVINO
+// path uses it to bind the embedding extractor to the correct port before inference.
+// Returns an error when the model has no 1024-dim output. The ONNX Runtime must be
+// initialized via InitONNXRuntime before calling.
+func DetectEmbeddingOutput(modelPath string) (index, size int, err error) {
+	return ort.DetectEmbeddingOutput(modelPath)
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/onnx.go:124]: Leaving this as-is for consistency. The onnx package documents ONNX Runtime initialization as a package-wide caller responsibility (see the runtime.go package doc), and none of the sibling constructors (NewClassifier, NewONNXCustomClassifier, NewRangeFilter) enforce it per call; they let the underlying ORT call surface the error. Adding an init guard only to DetectEmbeddingOutput would diverge from that convention. Its sole caller (NewBat) initializes the runtime before invoking it, and ort.GetInputOutputInfo already returns an error if the runtime is not initialized, so the failure is surfaced, just not with a bespoke message here.
